src/System.Private.ServiceModel/src/System/IdentityModel/Claims/X509CertificateClaimSet.cs

```diff
@@ -509,8 +514,13 @@ public X509Identity(X500DistinguishedName x500DistinguishedName)
         internal X509Identity(X509Certificate2 certificate, bool clone, bool disposable)
             : base(X509, X509)
         {
+#if TARGETS_WINDOWS
             _certificate = clone ? new X509Certificate2(certificate.Handle) : certificate;
             _disposable = clone || disposable;
+#else
```

Comment:
This overload is designed to allow a caller to specify whether to clone the certificate or not. I would prefer to modify the caller to use this overload and pass in false/false for clone/disposable.